ui/src/components/DeletePanelButton/DeletePanelButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useQueryClient } from "@tanstack/react-query";
+import { ConfirmationButton, useNotify } from "@canonical/react-components";
+import { Label, Props } from "./types";
+
+const DeletePanelButton: FC<Props> = ({
+  confirmButtonDisabled,
+  confirmButtonLabel,
+  confirmContent,
+  confirmTitle = "Confirm delete",
+  invalidateQuery,
+  entityName,
+  onDelete,
+  successMessage,
+  successPath,
+}) => {
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const [isLoading, setLoading] = useState(false);
+  const navigate = useNavigate();
+
+  const handleDelete = () => {
+    setLoading(true);
+    onDelete()
+      .then(() => {
+        navigate(successPath, notify.queue(notify.success(successMessage)));
+      })
+      .catch((e) => {
+        notify.failure(`${entityName} deletion failed`, e);
```

Comment:
question: is the `e` error type human readable enough for it to be displayed in the notification?

Reply:
That's a good question. This will display whatever is returned in the error `Message` param from the API, which at a quick look at the backend appears to be whatever internal error might occur, is that correct? If so I'll update this to not display those messages.

Reply:
In that case I would expect the template to be something like this
```
notify.failure(`${entityName} deletion failed`, e.Message ?? e);
```

Reply:
The message gets extracted by the API handler (https://github.com/canonical/identity-platform-admin-ui/blob/bffb816465023abfe249b21615b8bfdea5509848/ui/src/util/api.ts#L14 - lower case m from "Message string `json:"message"`" in the backend).

And the second param of `notify.failure()` takes an `Error` object which it then uses to extract the message: https://github.com/canonical/react-components/blob/83acef1f633fcf1a73641dca4287b8d89d944b96/src/components/NotificationProvider/messageBuilder.tsx#L41-L44.

However it looks like there was an issue with our code as depending on where the error comes from it might be an `Error` object or string so I've done some additional handling here to pass the message to the third param of `notify.failure()` if it's a string and added an extra test.

Reply:
Thank you @huwshimi , nice work!